datarepo-clienttests/src/main/java/common/utils/KubernetesClientUtils.java

```diff
@@ -66,6 +66,50 @@ public static AppsV1Api getKubernetesClientAppsObject() {
     return kubernetesClientAppsObject;
   }
 
+  /**
+   * Lock namespace - Throw exception if secret named "NAMESPACE-testrunner-inuse" exists If no
+   * existing secret, create a new secret
+   */
+  public static void lockNamespace(ServerSpecification server) throws Exception {
+    logger.info(
+        "Lock namespace by creating secret named '{}-testrunner-inuse'",
+        server.namespace,
+        server.namespace);
+    List<String> scriptArgs = new ArrayList<>();
+    scriptArgs.add("tools/lockNamespace.sh");
+    scriptArgs.add(server.clusterShortName);
+    scriptArgs.add(server.region);
+    scriptArgs.add(server.project);
+    scriptArgs.add(server.namespace);
+    Process fetchCredentialsProc = ProcessUtils.executeCommand("sh", scriptArgs);
+    List<String> cmdOutputLines = ProcessUtils.waitForTerminateAndReadStdout(fetchCredentialsProc);
+    if (fetchCredentialsProc.exitValue() > 0) {
+      throw new Exception("FAILURE: Failed to acquire lock for namespace " + server.namespace);
+    }
+    for (String cmdOutputLine : cmdOutputLines) {
+      logger.debug(cmdOutputLine);
+    }
+  }
+
+  /** unlock namespace - delete existing secret for "NAMESPACE-testrunner-inuse" */
+  public static void unlockNamespace(ServerSpecification server) throws Exception {
+    logger.info(
+        "unlock Env for namespace by deleting secret named '{}-testrunner-inuse'",
+        server.namespace,
+        server.namespace);
+    List<String> scriptArgs = new ArrayList<>();
+    scriptArgs.add("tools/unlockNamespace.sh");
+    scriptArgs.add(server.clusterShortName);
+    scriptArgs.add(server.region);
+    scriptArgs.add(server.project);
+    scriptArgs.add(server.namespace);
+    Process fetchCredentialsProc = ProcessUtils.executeCommand("sh", scriptArgs);
```

Comment:
I am not sure it is desirable that shell is driving the creation and deletion of secrets. I think that it may be useful to have the [Java code itself drive the locking and unlocking of secrets](
https://cloud.google.com/secret-manager/docs/creating-and-accessing-secrets#secretmanager-create-secret-java), especially since this is inherent to certain environments like perf. I suggest a method like `lockNamespace(ServerSpecification server, String lockName)` which would execute the locking step by creating a secret with the lock name. This could also potentially be reused by other environments which need this functionality.

The scripts are still useful for manual locking and unlocking.

Reply:
I agree that if we can do this in Java, that would be best (though the scripts are useful to have as reference). The helm commands use sh in a separate process only because I didn't find a well-supported Java client library to replace them with.

Reply:
From what I can tell with some testing, the Secret Manager API is different from the Kubernetes secrets. I think we should keep things consistent with the integration test locking mechanism (which uses Kubernetes secrets). But, it may be worth re-thinking how we lock clusters in both places.